include/zephyr/drivers/stepper.h

```diff
@@ -437,18 +437,18 @@ static inline int z_impl_stepper_is_moving(const struct device *dev, bool *is_mo
  * @retval 0 Success
  */
 __syscall int stepper_run(const struct device *dev, enum stepper_direction direction,
-			  uint32_t velocity);
+			  uint64_t step_interval_us);
 
 static inline int z_impl_stepper_run(const struct device *dev,
 				     const enum stepper_direction direction,
-				     const uint32_t velocity)
+				     const uint64_t step_interval_us)
```

Comment:
I see now that passing step interval again over here is just redundant. 
User would `set_interval`->`enable`->`run`
```suggestion
static inline int z_impl_stepper_run(const struct device *dev,
				     const enum stepper_direction direction)
```

Reply:
Yeah thats also true.

Reply:
But with this you can stop the motor  - maybe thats worth keeping?

Reply:
moved the stopping of the motor to the `set_step_interval` function.

Reply:
Yeah the idea is to decouple setting velocity from run. Run shall only do running in a given direction.